File: recursion/arraySum.cpp
#include<iostream>
using namespace std;

int arraySum(int arr[], int n){
  if(n==0) return 0;

  return arr[n-1] + arraySum(arr, n-1);
}

int main(){
int arr[] = {1,3,4,5,6,7};
int sum = arraySum(arr, 6);
cout<<sum<<endl;
return 0;
}

File: Recursion-js/arraySum.js
let arr = [3,2,6,7,1];
let sum=0;

function sumArr(n) {

    if(n<0) return 0;

    let current = (arr[n]%2!=0 ? arr[n]: 0);

    return current + sumArr(n-1);
}

 let ans = sumArr(arr.length-1);
console.log(ans);